### ISSUE TITLE
Marking a task as complete resets the task list to show all tasks

### ISSUE THREAD

**Issue by benjaminheng** [cs2103aug2015-f11-2j/main#54]
Marking a new task will cause previously marked tasks to appear.

**Steps to reproduce:**

```
add task1
add task2
add task3
mark 2
display
mark 2
```

**Problem:**
Before marking the second time, task1 and task3 are displayed. (correct)

```
task1
task3
```

After marking task3 as completed, task2 appears as well. (incorrect)

```
task1
task2 (complete)
task3 (complete)
```

**Expected result:**
After marking task3, the state should be:

```
task1
task3 (complete)
```

---

Probably because the code filters the master list instead of the displayed list?

**Comment by tanweijun** [comment]
I am aware of this issue, but is unable to think of a solution for it at the moment. It basically shows all tasks instead of the currently filtered tasks view after I mark something.

**Comment by benjaminheng** [comment]
Closed with wj's commit d1f4d068ee6e4cfa8f9bb5c48e71559622d6d370.
